File: root/branches/will/test-maintenance/source/modules/strategy/src/test/sample_data/inputs/part1.rb
include_class "org.marketcetera.strategy.ruby.Strategy"
include_class "java.lang.Long"
class Part1 < Strategy
  def on_start
      callbackCounter = get_property("onCallback")
      if(callbackCounter == nil)
          set_property("onCallback",
                       "0")
      end
      @helper = Helper.new
      @helper.do_something_common
      @helper.do_something_part1      
  end
  def on_ask(ask)
      set_property("ask" + self.to_s,
                   "part1")
  end
  def on_callback(data)
      callbackCounter = Long.parseLong(get_property("onCallback"))
      callbackCounter += 1
      set_property("onCallback",
                   Long.toString(callbackCounter))
      set_property("callback" + Long.toString(callbackCounter),
                   self.to_s)
  end
  def on_stop
      @helper.do_something_common
      @helper.do_something_part1
      @helper.do_something_part2
  end
end

class Helper
    def do_something_common
    end  
  def do_something_part1
  end
end

File: root/branches/toli/esper/source/modules/strategy/src/test/sample_data/inputs/Part1.java
import org.marketcetera.event.AskEvent;
import org.marketcetera.strategy.java.Strategy;

/* $License$ */

/**
 * Test strategy (part one of two parts) that tests what happens when a strategy helper-class is redefined. 
 *
 * @author <a href="mailto:colin@marketcetera.com">Colin DuPlantis</a>
 * @version $Id$
 * @since $Release$
 */
public class Part1
        extends Strategy
{
    private Helper helper;
    /* (non-Javadoc)
     * @see org.marketcetera.strategy.java.Strategy#onAsk(org.marketcetera.event.AskEvent)
     */
    @Override
    public void onAsk(AskEvent inAsk)
    {
        setProperty("ask" + this,
                    "part1");
    }
    /* (non-Javadoc)
     * @see org.marketcetera.strategy.java.Strategy#onCallback(java.lang.Object)
     */
    @Override
    public void onCallback(Object inData)
    {
        long callbackCounter = Long.parseLong(getProperty("onCallback"));
        callbackCounter += 1;
        setProperty("onCallback",
                    Long.toString(callbackCounter));
        setProperty("callback" + Long.toString(callbackCounter),
                    this.toString());
    }
    /* (non-Javadoc)
     * @see org.marketcetera.strategy.java.Strategy#onStart()
     */
    @Override
    public void onStart()
    {
        String callbackCounter = getProperty("onCallback");
        if(callbackCounter == null) {
            setProperty("onCallback",
                        "0");
        }
        helper = new Helper();
        helper.doSomethingCommon();
        helper.doSomethingPart1();      
    }
    /* (non-Javadoc)
     * @see org.marketcetera.strategy.java.Strategy#onStop()
     */
    @Override
    public void onStop()
    {
        helper.doSomethingCommon();
        helper.doSomethingPart1();
    }

    private static class Helper
    {
        private void doSomethingCommon()
        {
        }  
        private void doSomethingPart1()
        {
        }
    }
}